src/clims/services/extensible.py

```diff
@@ -1,11 +1,56 @@
 from __future__ import absolute_import
 
+import six
 from clims.services.substance import ExtensibleBaseField
 from clims.models import ExtensibleType, ExtensiblePropertyType
 from django.db import transaction
 
 
+class ExtensibleTypeNotRegistered(Exception):
+    pass
+
+
 class ExtensibleService(object):
+    def __init__(self, app):
+        self._app = app
+        self._implementation_cache = dict()
+        self._model_cache = dict()
```

Comment:
It's absolutely necessary. Otherwise we would have to go the DB to fetch type metadata everytime you're instantiating an extensible.

So an operation that everyone would think would be free, e.g. `sample = Sample()` would lead to just about the largest overhead possible, a call to the DB.

Reply:
Good catch about clearing the implementation cache in clear cache. That was a mistake. I'll rename it to implementations.

Reply:
I think you are right about the query going to the database. Originally I was thinking that this might be cached by the ORM layer, but I read up on that now, and that does not seem to be the case. I'm mostly asking about it since I think that it looks a little bit like a premature optimization (and caches are often difficult to reason about). However, I good with keeping it this way. :+1:

Reply:
Of course knowing what's premature and what's not is an art, but I would say that a good rule of thumb is never to go to the DB without a reason. A better example of premature opt in our context would be more on the level of "should I go to the hard disk on the server or just to memory".

The downside of not optimizing obvious stuff right away is that then you miss out on months of testing of that feature. My view is that if it should be there and it doesn't take much time to add it, it should go in right away (getting-things-done-style).

Reply:
:+1: